### ISSUE TITLE
"gperf-3.1" & "Libelf from Elfutils-0.180" not installed

### ISSUE THREAD

**Issue by ahmdprog** [luisgbm/lfs-scripts#1]
Hello,

Thank you for scripts provided. using your script and after installation "gperf" and "Libelf from Elfutils-0.180" is not installed for some reason. i tried to figure out why but I couldn't.

**Comment by luisgbm** [comment]
Hi there! Thanks for your feedback. Could you please provide the log file for the execution of the lfs-system.sh script? The components you mentioned are installed in this phase, and maybe with the logs we'll be able to see what went wrong.

**Comment by ahmdprog** [comment]
Hello, as you can see the below logs, the script did not install anything. can you please check your testing environment if gperf-3.1 installed ?

[lfs-scripts] Starting build of gperf-3.1 at Sun Jan 24 02:06:40 -03 2021
checking whether make sets $(MAKE)... yes
configure: creating ./config.status
config.status: creating Makefile
=== configuring in lib (/sources/gperf-3.1/lib)
configure: running /bin/sh ./configure --disable-option-checking '--prefix=/usr'  '--docdir=/usr/share/doc/gperf-3.1' --cache-file=/dev/null --srcdir=.
checking whether make sets $(MAKE)... yes
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ISO C89... none needed
checking how to run the C preprocessor... gcc -E
checking for g++... g++
checking whether we are using the GNU C++ compiler... yes
checking whether g++ accepts -g... yes
checking how to run the C++ preprocessor... g++ -E
checking for ar... ar
checking for ranlib... ranlib
checking for a BSD compatible install... /usr/bin/install -c
configure: creating ./config.status
config.status: creating Makefile
=== configuring in src (/sources/gperf-3.1/src)
configure: running /bin/sh ./configure --disable-option-checking '--prefix=/usr'  '--docdir=/usr/share/doc/gperf-3.1' --cache-file=/dev/null --srcdir=.
checking whether make sets $(MAKE)... yes
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ISO C89... none needed
checking how to run the C preprocessor... gcc -E
checking for g++... g++
checking whether we are using the GNU C++ compiler... yes
checking whether g++ accepts -g... yes
checking how to run the C++ preprocessor... g++ -E
checking for a BSD compatible install... /usr/bin/install -c
checking for stack-allocated variable-size arrays... yes
checking for rand in -lm... yes
configure: creating ./config.status
config.status: creating Makefile
config.status: creating config.h
=== configuring in tests (/sources/gperf-3.1/tests)
configure: running /bin/sh ./configure --disable-option-checking '--prefix=/usr'  '--docdir=/usr/share/doc/gperf-3.1' --cache-file=/dev/null --srcdir=.
checking whether make sets $(MAKE)... yes
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ISO C89... none needed
checking how to run the C preprocessor... gcc -E
checking for g++... g++
checking whether we are using the GNU C++ compiler... yes
checking whether g++ accepts -g... yes
checking how to run the C++ preprocessor... g++ -E
configure: creating ./config.status
config.status: creating Makefile
=== configuring in doc (/sources/gperf-3.1/doc)
configure: running /bin/sh ./configure --disable-option-checking '--prefix=/usr'  '--docdir=/usr/share/doc/gperf-3.1' --cache-file=/dev/null --srcdir=.
checking whether make sets $(MAKE)... yes
checking for a BSD compatible install... /usr/bin/install -c
configure: creating ./config.status
config.status: creating Makefile
cd lib; make all
make[1]: Entering directory '/sources/gperf-3.1/lib'
gcc -g -O2 -I. -c ./getopt.c
gcc -g -O2 -I. -c ./getopt1.c
g++ -g -O2 -I. -c ./getline.cc
g++ -g -O2 -I. -c ./hash.cc
rm -f libgp.a
ar rc libgp.a getopt.o getopt1.o getline.o hash.o
ranlib libgp.a
make[1]: Leaving directory '/sources/gperf-3.1/lib'
cd src; make all
make[1]: Entering directory '/sources/gperf-3.1/src'
g++ -g -O2 -I. -I./../lib -c ./version.cc
g++ -g -O2 -I. -I./../lib -c ./positions.cc
g++ -g -O2 -I. -I./../lib -c ./options.cc
g++ -g -O2 -I. -I./../lib -c ./keyword.cc
g++ -g -O2 -I. -I./../lib -c ./keyword-list.cc
g++ -g -O2 -I. -I./../lib -c ./input.cc
g++ -g -O2 -I. -I./../lib -c ./bool-array.cc
g++ -g -O2 -I. -I./../lib -c ./hash-table.cc
g++ -g -O2 -I. -I./../lib -c ./search.cc
g++ -g -O2 -I. -I./../lib -c ./output.cc
g++ -g -O2 -I. -I./../lib -c ./main.cc
make[1]: Leaving directory '/sources/gperf-3.1/src'
cd lib; make check
make[1]: Entering directory '/sources/gperf-3.1/lib'
make[1]: Nothing to be done for 'check'.
make[1]: Leaving directory '/sources/gperf-3.1/lib'
cd src; make check
make[1]: Entering directory '/sources/gperf-3.1/src'
g++ -g -O2 -I. -I./../lib -c ./positions.cc
make[1]: Leaving directory '/sources/gperf-3.1/src'
cd lib; make install
make[1]: Entering directory '/sources/gperf-3.1/lib'
make[1]: Nothing to be done for 'install'.
make[1]: Leaving directory '/sources/gperf-3.1/lib'
cd src; make install
make[1]: Entering directory '/sources/gperf-3.1/src'
g++ -g -O2 -I. -I./../lib -c ./positions.cc
g++ -g -O2 -I. -I./../lib -c ./options.cc
g++ -g -O2 -I. -I./../lib -c ./keyword.cc
g++ -g -O2 -I. -I./../lib -c ./input.cc
g++ -g -O2 -I. -I./../lib -c ./search.cc
g++ -g -O2 -I. -I./../lib -c ./main.cc
make[1]: Leaving directory '/sources/gperf-3.1/src'
[lfs-scripts] Finishing build of gperf-3.1 at Sun Jan 24 02:06:43 -03 2021

**Comment by luisgbm** [comment]
Hi, sorry I took so long to respond. Your log seems correct, and if you check the official build logs provided by the LFS team, it matches: http://www.linuxfromscratch.org/lfs/build-logs/10.0/i5-6600K/logs/835-gperf-3.1

The fact that it says "Nothing to be done for 'install'" seems to be normal. What you really need to check is if your final LFS build has the "gperf" command available. I've checked here in my environment, and it works (in this screenshot I ran the "man gperf" command, but "gperf" also works):

![gperf](https://user-images.githubusercontent.com/702411/106622274-2fe20f80-6552-11eb-9a92-c36b6e6e2f10.png)

As for libelf, the lib files seems to be correctly installed in the /lib directory:

![libelf](https://user-images.githubusercontent.com/702411/106622358-40928580-6552-11eb-8cf8-bd83cdd68553.png)

I'm closing this issue for now, feel free to contact me if you have further questions. Thanks!
